src/EFCore.PG.NodaTime/NodaTimeMethodCallTranslator.cs

```diff
@@ -45,11 +46,41 @@ public class NodaTimeMethodCallTranslator : IMethodCallTranslator
         [NotNull] static readonly MethodInfo GetCurrentInstant =
             typeof(SystemClock).GetRuntimeMethod(nameof(SystemClock.GetCurrentInstant), Type.EmptyTypes);
 
+        /// <summary>
+        /// The mapping of supported method translations.
+        /// </summary>
+        [NotNull] static readonly Dictionary<MethodInfo, string> MethodInfoDatePartMapping = new Dictionary<MethodInfo, string>
+        {
+            { typeof(Period).GetRuntimeMethod(nameof(Period.FromYears),        new[] { typeof(int) }),  "years" },
```

Comment:
Just a nit, but using the `[key] = value` syntax here (as you do below) might be a bit more terse.

Reply:
I gave it a shot, but having the huge MethodImpl expression inside those square brackets didn't make things any better for me visually :)